python/tvm/script/parser.py

```diff
@@ -276,6 +280,22 @@ def parse_arg_list(self, func, node_call):
         reader = CallArgumentReader(func_name, args, kw_args, self, node_call)
         pos_only, kwargs, varargs = param_list
         internal_args = list()
+        # check the default value of "name" for TypeCall and TypeApply
+        # if none, change it to the variable name
+        # Note that here the variable name will always be the last
+        # item in self.context.func_params. See transform_Function
+        # for more details of adding function parameters in self.context
+        buf_name: str = self.context.func_params[-1].name
+        if isinstance(node_call, ast.TypeApply):
+            if len(args) == 2:
```

Comment:
Rather than checking `len(args)` here we should use more explicit and clear way to check whether buffer name is present. One possible way is to force users to use kwargs for parameters other than `shape` and `dtype` so that you can check and set kw_args here